nodes/voice_teleop

```diff
@@ -351,6 +358,24 @@ class KeyboardTeleopNode(hm.HelloNode):
             rospy.loginfo('Done sending pose.')
      #   self.topIntent = None
 
+    # If some specific buttons are pressed on XBox controller, that would mean user is requesting for a hard stop
+    def check_for_hardstop(self):
+        hardstop_triggered = False
+        xbox_controller_state = self.xbox_controller.get_state()
+        for k in xbox_controller_state.keys():
+            if (k == "start_button_pressed" or k == "select_button_pressed" or k == "top_button_pressed" or k == "left_button_pressed" or
+                k == "bottom_button_pressed" or k == "right_button_pressed") and xbox_controller_state[k] == True:
+
+                print(k, ' : ', True)
+                hardstop_triggered = True
+
+        return hardstop_triggered
+
+    # Initiate kill switch on the Robot
+    def initiate_hardstop(self):
+        self.trigger_stop_the_robot = rospy.ServiceProxy('/stop_the_robot', Trigger)
```

Comment:
Is this the right way to initiate stop?

Reply:
```suggestion
        self.trigger_stop_the_robot = rospy.ServiceProxy('/stop_the_robot', Trigger)
```
How about adding it this way as in Binit's example?
        self.stop_the_robot_service = rospy.Service('/stop_the_robot',
                                                    Trigger,
                                                    self.stop_the_robot_callback)

        self.runstop_service = rospy.Service('/runstop',
                                              SetBool,
                                              self.runstop_service_callback)

Reply:
Thanks @Nikhi123 
I understood the mistake in parameter i.e. I should have used SetBool, will do that change

But, do i need to add the callback function as well?

Reply:
I think one more difference is that I am using ServiceProxy instead of Service.
@binitshah what do you suggest?

Reply:
@japsingh sure :) and no callback according to me for your changes. SetBool should do it

Reply:
Done